Doc/library/asyncio-stream.rst

```diff
@@ -206,12 +206,20 @@ StreamReader
 
    .. coroutinemethod:: read(n=-1)
 
-      Read up to *n* bytes.  If *n* is not provided, or set to ``-1``,
-      read until EOF and return all read bytes.
+      Read up to *n* bytes.
 
+      If *n* is not provided, or set to ``-1``,
+      read until EOF. After EOF is received, return all read bytes.
       If EOF was received and the internal buffer is empty,
       return an empty ``bytes`` object.
 
+      If *n* is zero, return empty ``bytes`` object immediately.
+
+      If *n* is positive, this function tries to read *n* bytes, and may return
+      less or equal bytes than requested, but at least one byte. If EOF was
+      received before any byte is read, this function returns an empty
+      ``bytes`` object.
```

Comment:
```suggestion
      If *n* is positive, read at most *n* bytes.
      May return less than *n* bytes, if EOF is reached first.
      If EOF is received before any byte is read,
      return an empty ``bytes`` object.
```

* Specify the actual reason for less than *n* bytes being returned—EOF is reached first (assuming I'm interpreting this correctly, and it parallels the ``-1`` case).
* Avoid apparent contradictory statement that at least one byte will be returned, yet can also return empty bytes if EOF is received first
* Split first sentence into two, for clarity and readability
* Elide redundant (and imprecise) "this function" verbiage, and be consistent with the phrasing used elsewhere here

Reply:
> Specify the actual reason for less than n bytes being returned—EOF is reached first (assuming I'm interpreting this correctly, and it parallels the -1 case).

This is not correct. It can also return less than `n` bytes, if all `n` bytes are not immediately available (but at least 1 byte). This was the cause of a bug in one of my Python programs (less than `n` bytes because EOF would have been handled fine).

Reply:
Indeed.

Reply:
I pushed which I hope incorporates your suggestions but keeps the correct reasons less than `n` bytes might be returned.

Reply:
I've addressed further refinements in a followup suggestion, which was applied.